### PR TITLE
Implement lakectl local list

### LINKED ISSUE #6281: Implement lakectl local list
Parent: https://github.com/treeverse/lakeFS/issues/6239

`lakectl local list`

Find all lakeFS-synced directories (i.e. directories with a .lakefs_ref.yaml at their root) in the current directory - or if in a Git repository: from the root of the repository.

### PR DESCRIPTION
**Pull request by N-o-Z**
### Linked Issue

Closes #6281

---

### Background

Implement lakectl local command which lists all folders linked to lakefs paths
      
### New Feature

Part of task https://github.com/treeverse/lakeFS/issues/6241

### Testing Details

Unit testing for tools, lakectl tests will be added for esti upon feature completion

### Breaking Change?

No

---

### CODE REVIEW

**Review by nopcoder — CHANGES_REQUESTED** [review]
Added some comments about the code.
What is the use case of this command?
Was expected that we will have a command that will accept or use the current directory and lookup (in path or parents) for our yaml file and dump the information. Command like: lakectl local show.
Using find (command like utility) and lakectl local show I can run something like this list command.

File: cmd/lakectl/cmd/local_list.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+			rows = append(rows, table.Row{d, remote, idx.AtHead})
+		}
+		data := struct {
+			IndicesListTable *Table
+		}{
+			IndicesListTable: &Table{
+				Headers: []interface{}{
+					"Directory",
+					"Remote URI",
+					"Last synced HEAD",
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply cmd/lakectl/cmd/local_list.go:64]: I think the ref is too general. Should be either head or commit. WDYT?

File: pkg/git/git.go [hunk 2]
```diff
@@ -21,6 +21,10 @@ const (
 )
 
 func git(dir string, args ...string) (string, error) {
+	_, err := exec.LookPath("git") // assume git is in path, otherwise consider as not having git support
+	if err != nil {
+		return "", ErrNoGit
+	}
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/git/git.go:27]: There are 2 types of errors we would like to specifically distinguish from which are the environment doesn't support git, and the current working directory is not a git repository.

**Comment by N-o-Z** [comment]
> Added some comments about the code. What is the use case of this command? Was expected that we will have a command that will accept or use the current directory and lookup (in path or parents) for our yaml file and dump the information. Command like: lakectl local show. Using find (command like utility) and lakectl local show I can run something like this list command.

This is basically the same functionality you described. When given a linked directory it will act exactly as 'show', and when providing a parent dir it will find all linked directories under that path

File: pkg/git/git.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nopcoder) [reply pkg/git/git.go:27]: You added lookup for git - so in the case we found git - we left only with the fact that that we have git tool and it didn't recognize a git repository.

File: pkg/local/index.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	if idxPath == "" {
+		return nil, fmt.Errorf("%s: %w", path, fs.ErrNotExist)
+	}
+	data, err := os.ReadFile(idxPath)
+	if err != nil {
+		return nil, err
+	}
+	idx := &Index{}
+	err = yaml.Unmarshal(data, idx)
+	idx.root = filepath.Dir(idxPath)
```
Comment (nopcoder) [inline pkg/local/index.go:78]: move after the err check

**Review by nopcoder — APPROVED** [review]
LGTM - I'll move my functional comment to the design document.
Added comment in `ReadIndex` and optional suggestion for the column name.

File: pkg/git/git.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/git/git.go:27]: Remove the generic GitError

**Pull request merged**